packages/styling/src/utilities/icons.ts

```diff
@@ -31,8 +31,21 @@ export interface IIconRecord {
 }
 
 export interface IIconOptions {
+  /**
+   * By default, registering the same set of icons will generate a console warning per duplicate icon
+   * registered, because this scenario can create unexpected consequences.
+   *
+   * Some scenarios include:
+   *
+   * Icon set was previously registered using a different base url.
+   * Icon set was previously registered but a different version was provided.
+   * Icons in a previous registered set overlap with a new set.
+   *
+   * To simply ignore previously registered icons, you can specify to disable warnings. This means
+   * that if an icon which was previous registered is registered again, it will be silently ignored.
+   * However, consider whether the problems listed above will cause issues.
+   **/
   disableWarnings: boolean;
-  warnOnMissingIcons: boolean;
```

Comment:
Is there a reason to remove the parameter, rather than deprecate? Isn't this an API break?

Reply:
I added it back to ensure backwards compatibility.